File: app/src/main/java/fr/frogdevelopment/pregnancycalendar/data/LocalDateTimeConverter.java
package fr.frogdevelopment.pregnancycalendar.data;

import androidx.room.TypeConverter;

import java.time.Instant;
import java.time.ZoneId;
import java.time.ZonedDateTime;

public class LocalDateTimeConverter {

    @TypeConverter
    public static ZonedDateTime toDate(Long date) {
        if (date == null) {
            return null;
        } else {
            return Instant.ofEpochSecond(date).atZone(ZoneId.systemDefault());
        }
    }

    @TypeConverter
    public static Long toSqlDate(ZonedDateTime date) {
        if (date == null) {
            return null;
        } else {
            return date.toEpochSecond();
        }
    }
}

File: app/src/main/java/fr/frogdevelopment/pregnancycalendar/data/LocalDateTimeConverter.kt
package fr.frogdevelopment.pregnancycalendar.data

import androidx.room.TypeConverter
import org.threeten.bp.Instant
import org.threeten.bp.ZoneId
import org.threeten.bp.ZonedDateTime

object LocalDateTimeConverter {
    @JvmStatic
    @TypeConverter
    fun toDate(date: Long?): ZonedDateTime? {
        return if (date == null) {
            null
        } else {
            Instant.ofEpochSecond(date).atZone(ZoneId.systemDefault())
        }
    }

    @JvmStatic
    @TypeConverter
    fun toSqlDate(date: ZonedDateTime?): Long? {
        return date?.toEpochSecond()
    }
}